src/analysis/rust_type.rs

```diff
@@ -347,6 +347,12 @@ impl<'env> RustTypeBuilder<'env> {
                 skip_option = true;
                 let inner_ref_mode = match self.env.library.type_(inner_tid) {
                     Class(..) | Interface(..) => RefMode::None,
+                    Record(..) => {
+                        match RecordType::of(self.env.type_(inner_tid).maybe_ref().unwrap()) {
+                            RecordType::AutoBoxed | RecordType::Boxed => RefMode::None,
+                            _ => self.ref_mode,
```

Comment:
I think this can use some more auto-detection here. If we generate an `BoxedInline` then it should do this automatically, otherwise based on `ref_mode` configuration or fall back to the default.

Reply:
Hmm but not all `BoxedInline` are `Copy`. Do we have a way to detect this better?

Reply:
> Hmm but not all `BoxedInline` are `Copy`. Do we have a way to detect this better?

I updated the code to handle that. 

> I think this can use some more auto-detection here. If we generate an BoxedInline then it should do this automatically, otherwise based on ref_mode configuration or fall back to the default.

Not sure what kind of auto-detection that would be needed here?

Reply:
Seems fine for now I guess. If we need more then we can add that later